Update syntax paper with result of plenary discussion (#7)
j3-fortran/async-coll: drafts/draft-syntax.txt

Subroutines" presents an illustrative use case and a list of requirements.

Paper J3/25-165r1 "US04: Specifications for Asynchronous Collective
Subroutines" presents the specifications and semantics for the new
asynchronous collective subroutines feature. It passed at the Oct 2025 J3
meeting #237.  Please consult that paper for all semantic details, which
will not be replicated here.

The current paper focuses exclusively on the proposed syntactic elements
for the asynchronous collective subroutines feature.  This revision of the
paper incorporates J3 committee feedback provided in discussion of revision
J3/25-166r1 on 2025-10-22, and applies the results from a series of straw

> polls used to settle the syntax.

Author: Do we need to record the detailed results of each straw poll in the paper, or is the statement above good enough?

Author: Brandon and I both believe this is fine as-is, given the results of the straw polls should appear in the meeting minutes for anyone who's really interested.

3. Asynchronous Collective Initiation Syntax
--------------------------------------------

Paper J3/25-127r1 "DIN1: Collectives over a specified team,
req/spec/syntax/edits" (passed at Jun 2025 J3 meeting #236), adds a new
TEAM argument to the same collective subroutines impacted by work item

> US04.  These two features (US04 and DIN1) are entirely orthogonal, but

Reviewer: ```suggestion
US04.  The functionality of these two features (US04 and DIN1) is entirely orthogonal, but
```

their effects must be merged in the final syntax.

For clarity of direction, this paper incorporates the syntactic changes
from work item DIN1, and supersedes the syntax presented in section 4 of
J3/25-127r1.  A forthcoming edit paper is also expected to supersede the
edits passed in section 5 of J3/25-127r1.

Each of the collective subroutines are expanded into four forms,
corresponding to the appearance or non-appearance of a COMPLETION argument

> of type COMPLETION_TYPE and a TEAM argument of type TYPE_TYPE (also from

Author: ```suggestion
of type COMPLETION_TYPE and a TEAM argument of type TEAM_TYPE (also from
```

    CO_MAX (A [, RESULT_IMAGE], TEAM [, STAT, ERRMSG]) or
    CO_MAX (A [, RESULT_IMAGE], TEAM, COMPLETION [, STAT, ERRMSG])

C3. CO_MIN (A [, RESULT_IMAGE, STAT, ERRMSG]) or
    CO_MIN (A [, RESULT_IMAGE], COMPLETION [, STAT, ERRMSG]) or
    CO_MIN (A [, RESULT_IMAGE], TEAM [, STAT, ERRMSG]) or
    CO_MIN (A [, RESULT_IMAGE], TEAM, COMPLETION [, STAT, ERRMSG])

C4. CO_REDUCE (A, OPERATION [, RESULT_IMAGE, STAT, ERRMSG]) or
    CO_REDUCE (A, OPERATION [, RESULT_IMAGE], COMPLETION
                                              [, STAT, ERRMSG]) or
    CO_REDUCE (A, OPERATION [, RESULT_IMAGE], TEAM 
                                              [, STAT, ERRMSG]) or
    CO_REDUCE (A, OPERATION [, RESULT_IMAGE], TEAM, COMPLETION
                                              [, STAT, ERRMSG])

C5. CO_SUM (A [, RESULT_IMAGE, STAT, ERRMSG]) or
    CO_SUM (A [, RESULT_IMAGE], COMPLETION [, STAT, ERRMSG]) or
    CO_SUM (A [, RESULT_IMAGE], TEAM [, STAT, ERRMSG]) or
    CO_SUM (A [, RESULT_IMAGE], TEAM, COMPLETION [, STAT, ERRMSG])

See J3/25-165r1 for detailed specifications and semantics; since that
paper, the COMPLETION argument has been changed from an optional argument

> to a required argument with alternative intrinsics forms. However the same

Author: ```suggestion
to a required argument with alternative intrinsic forms. However the same
```
